src/parallel/GeneralDomainDecomposition.cpp

```diff
@@ -13,26 +13,38 @@
 #include "NeighbourCommunicationScheme.h"
 
 GeneralDomainDecomposition::GeneralDomainDecomposition(double interactionLength, Domain* domain)
-	: _boxMin{0.}, _boxMax{0.} {
-	std::array<double, 3> domainLength = {domain->getGlobalLength(0), domain->getGlobalLength(1),
-										  domain->getGlobalLength(2)};
-
-	auto gridSize = getOptimalGrid(domainLength, this->getNumProcs());
+	: _boxMin{0.},
+	  _boxMax{0.},
+	  _domainLength{domain->getGlobalLength(0), domain->getGlobalLength(1), domain->getGlobalLength(2)},
+	  _interactionLength{interactionLength} {}
+
+void GeneralDomainDecomposition::initializeALL() {
+	global_log->info() << "initializing ALL load balancer..." << std::endl;
+	auto gridSize = getOptimalGrid(_domainLength, this->getNumProcs());
 	auto gridCoords = getCoordsFromRank(gridSize, _rank);
-	_coversWholeDomain = {gridSize[0] == 1, gridSize[1] == 1, gridSize[2] == 1};
 	global_log->info() << "gridSize:" << gridSize[0] << ", " << gridSize[1] << ", " << gridSize[2] << std::endl;
 	global_log->info() << "gridCoords:" << gridCoords[0] << ", " << gridCoords[1] << ", " << gridCoords[2] << std::endl;
-	std::tie(_boxMin, _boxMax) = initializeRegularGrid(domainLength, gridSize, gridCoords);
+	std::tie(_boxMin, _boxMax) = initializeRegularGrid(_domainLength, gridSize, gridCoords);
+	if (_gridSize) {
+		std::tie(_boxMin, _boxMax) = latchToGridSize(_boxMin, _boxMax);
+	}
 #ifdef ENABLE_ALLLBL
 	// Increased slightly to prevent rounding errors.
-	double minimalDomainSize{interactionLength * (1. + 1.e-10)};
+	double safetyFactor = 1. + 1.e-10;
+	std::array<double, 3> minimalDomainSize{_interactionLength * safetyFactor, _interactionLength * safetyFactor,
+											_interactionLength * safetyFactor};
+	if (_gridSize) {
+		for (size_t i = 0; i < 3; ++i) {
+			minimalDomainSize[i] = (*_gridSize)[i];
+		}
+	}
```

Comment:
```suggestion
	double safetyFactor = 1. + 1.e-10;
	std::array<double, 3> minimalDomainSize = _gridSize ?
	  {(*_gridSize)[0], (*_gridSize)[1], (*_gridSize)[2]} :
	  {_interactionLength * safetyFactor, _interactionLength * safetyFactor, _interactionLength * safetyFactor};
```

Or initialize both versions through the loop with an if within the loop but the current mixed form look confusing.